### ISSUE TITLE
isEmpty triggers false-positive when using with XCUIElementQuery

### ISSUE THREAD

**Issue by christiankm** [realm/SwiftLint#5182]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When trying to compare the cell count in a table from XCUITests, the isEmpty rule triggers, when we're using code like .cells.count > 0.

Problem is, that XCUIElementQuery.cells is not an array and has no isEmpty property, hence the false positive.

I would consider this a bug in SwiftFormat for now, though it would be more appropriate to add an isEmpty property so we can use that, to the XCTest framework.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.52.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
XCTAssertTrue(app.tables["table"].cells.count > 0)
```

<img width="1149" alt="Skærmbillede 2023-08-15 kl  18 19 09" src="https://github.com/realm/SwiftLint/assets/3626094/947cf0f2-5c57-46e2-ac23-ff028cb0d5ee">

**Comment by marcelofabri** [comment]
SwiftLint doesn't have enough information to tell if the underlying object has an `isEmpty` method - that's the reason why this rule isn't enabled by default. 

See https://github.com/realm/SwiftLint/issues/827 for more discussion.
